### ISSUE TITLE
Declarative way of mapping data

### ISSUE THREAD

**Issue by webberwang** [codelab-ai/codelab.ai#177]
Test drive design, implement Mapper so test passes

```typescript
// libs/shared/utils/src/mapper.spec.ts
class Mapper {
  constructor(original: any, map: any) {
    console.log()
  }

  execute() {
    console.log()
  }
}

describe('Mapper', () => {
  it('maps from one shape to another', () => {
    const original = {
      data: {
        address: {
          street_1: '2600 Park Ave.',
          street_2: '',
          city: 'New York',
        },
      },
    }

    const expected = {
      address: {
        street: '2600 Park Ave.',
        city: 'New York',
      },
    }

    const map = [
      ['data.address.street_1', 'address.street'],
      ['data.address.city', 'address.city'],
    ]

    const mapper = new Mapper(original, map)

    expect(mapper.execute()).toStrictEqual(expected)
  })
})
```

**Comment by webberwang** [comment]
Closed by #223

**Comment by VienNguyen24hdev** [comment]
via Hubstaff
User: Nguyen Luong Vien

Project: Codelab - https://app.hubstaff.com/projects/1093588
Date Range: 10/15/20 - 10/16/20
Work session total: 1:47:18
Billable: Yes

Grand total: 1:47:18
